### PR TITLE
[JENKINS-35328] Bump BouncyCastle version

### PR DESCRIPTION
**Pull request by armfergom**
[JENKINS-35328](https://issues.jenkins-ci.org/browse/JENKINS-35328)

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -339,8 +355,8 @@
     </dependency>
     <dependency>
       <groupId>org.bouncycastle</groupId>
-      <artifactId>bcprov-jdk16</artifactId>
-      <version>1.46</version>
+      <artifactId>bcpkix-jdk15on</artifactId>
+      <version>1.54</version>
```
Comment (armfergom) [inline pom.xml]: `bcpkix-jdk15on` includes `bcprov-jdk16`

**Comment by alvarolobato** [comment]
🐝

**Comment by abayer** [comment]
:bee:

**Comment by armfergom** [comment]
@reviewbybees done

**Pull request merged**